File: core/src/main/java/org/dbpedia/extraction/mappings/rml/RMLParser.java
package org.dbpedia.extraction.mappings.rml;

import be.ugent.mmlab.rml.mapdochandler.extraction.std.StdRMLMappingFactory;
import be.ugent.mmlab.rml.mapdochandler.retrieval.RMLDocRetrieval;
import be.ugent.mmlab.rml.model.RMLMapping;
import org.openrdf.repository.Repository;
import org.openrdf.rio.RDFFormat;

/**
 * Responsible for parsing RML documents using the MapDocHandler
 */
public class RMLParser {

    private RMLDocRetrieval retriever;
    private StdRMLMappingFactory rmlMappingFactory;

    public RMLParser() {
        retriever = new RMLDocRetrieval();
        rmlMappingFactory = new StdRMLMappingFactory();
    }

    /**
     * Parses RML document from a file and returns an RMLMapping object
     */
    public RMLMapping parseFromFile(String pathToRmlDocument) {

        Repository repo = retriever.getMappingDoc(pathToRmlDocument, RDFFormat.TURTLE);
        Repository preparedRepo = rmlMappingFactory.prepareExtractRMLMapping(repo);
        RMLMapping rmlMapping = rmlMappingFactory.extractRMLMapping(preparedRepo);
        return rmlMapping;

    }



}

File: core/src/main/scala/org/dbpedia/extraction/mappings/rml/RMLParser.scala
package org.dbpedia.extraction.mappings.rml
import java.nio.file
import java.nio.file.{Path, Paths}

import be.ugent.mmlab.rml.mapdochandler.extraction.std.StdRMLMappingFactory
import be.ugent.mmlab.rml.mapdochandler.retrieval.RMLDocRetrieval
import be.ugent.mmlab.rml.model.RMLMapping
import org.openrdf.rio.RDFFormat

/**
  * Responsible for parsing RML documents using the MapDocHandler
  */


object RMLParser {

  private val retriever : RMLDocRetrieval = new RMLDocRetrieval()
  private val rmlMappingFactory : StdRMLMappingFactory = new StdRMLMappingFactory()

  /**
    * Parses RML document from a file and returns an RMLMapping object
    */
  def parseFromFile(pathToRmlDocument: String): RMLMapping = {

      val path = convertToAbsolutePath(pathToRmlDocument)
      val repo = retriever.getMappingDoc(path, RDFFormat.TURTLE)
      val preparedRepo = rmlMappingFactory.prepareExtractRMLMapping(repo)
      val rmlMapping = rmlMappingFactory.extractRMLMapping(preparedRepo)

      rmlMapping

  }

  /**
    * Converts path to absolute path
    */
  private def convertToAbsolutePath(path: String): String = {
      val relativePath = Paths.get(path)
      val absolutePath = relativePath.toAbsolutePath
      return absolutePath.toString
  }

}
